### ISSUE TITLE
knowledge distillation training seems broken

### ISSUE THREAD

**Issue by LouieShao** [zgcr/SimpleAICV-pytorch-ImageNet-COCO-training#23]
First, thanks author for the great work. It is a great tool to conduct ablation studies. (I even think you can write a paper about that, after adding some more training option, e.g., few-shot, zero-shot learning.)

However, there seems to have a bug about distillation training. That is, when i finished downloading ResNet-34 weights, loading them to the teacher model, the accuracy of it seems really low. It claims 0.298% top-1 accuracy on ImageNet-1K. 

I did not checked the code yet, but i suspect it is because the weights you published has a different order of output classes. Could you kindly check this out?

**Comment by zgcr** [comment]
> First, thanks author for the great work. It is a great tool to conduct ablation studies. (I even think you can write a paper about that, after adding some more training option, e.g., few-shot, zero-shot learning.)
> 
> However, there seems to have a bug about distillation training. That is, when i finished downloading ResNet-34 weights, loading them to the teacher model, the accuracy of it seems really low. It claims 0.298% top-1 accuracy on ImageNet-1K.
> 
> I did not checked the code yet, but i suspect it is because the weights you published has a different order of output classes. Could you kindly check this out?

Hi，Please theck out your ResNet-34 weight from resnet folder,not resnet_norm_div_255 folder.
load_state_dict function in simpleAICV/classification/common.py is used to load model weight(loading process actually happend in KDModel class in train_config.py),and this function has been considered the change of class_num.A layer shape will be loaded when model and weight has same weight layer name and same weight layer shape.If you suspect the weight is not loaded correctly,you can test this weight by using test_config.py and test.sh.

**Comment by zgcr** [comment]
> First, thanks author for the great work. It is a great tool to conduct ablation studies. (I even think you can write a paper about that, after adding some more training option, e.g., few-shot, zero-shot learning.)
> 
> However, there seems to have a bug about distillation training. That is, when i finished downloading ResNet-34 weights, loading them to the teacher model, the accuracy of it seems really low. It claims 0.298% top-1 accuracy on ImageNet-1K.
> 
> I did not checked the code yet, but i suspect it is because the weights you published has a different order of output classes. Could you kindly check this out?

Another reason is that the weights are not loaded correctly because the acc is very close to the level at initialization model.May be the weight didn't load correctly(wrong dir?)?

**Comment by LouieShao** [comment]
Thanks for your quick response. I will check and report the result asap.

**Comment by LouieShao** [comment]
I found where is wrong. It is because I generate folder names compared with the default one. Thanks for your help. I will close this issue.
